pubsub/google/cloud/pubsub_v1/publisher/batch/base.py

```diff
@@ -0,0 +1,171 @@
+
+        Returns:
+            Sequence: The messages currently in the batch.
+        """
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def size(self):
+        """Return the total size of all of the messages currently in the batch.
+
+        Returns:
+            int: The total size of all of the messages currently
+                 in the batch, in bytes.
+        """
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def settings(self):
+        """Return the batch settings.
+
+        Returns:
+            ~.pubsub_v1.types.BatchSettings: The batch settings. These are
+                considered immutable once the batch has been opened.
+        """
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def status(self):
+        """Return the status of this batch.
+
+        Returns:
+            str: The status of this batch. All statuses are human-readable,
+                all-lowercase strings. The ones represented in the
+                :class:`BaseBatch.Status` enum are special, but other statuses
+                are permitted.
+        """
+        raise NotImplementedError
+
+    def will_accept(self, message):
+        """Return True if the batch is able to accept the message.
+
+        Args:
+            message (~.pubsub_v1.types.PubsubMessage): The Pub/Sub message.
+
+        Returns:
+            bool: Whether this batch can accept the message.
+        """
+        # If this batch is not accepting messages generally, return False.
+        if self.status != self.Status.ACCEPTING_MESSAGES:
+            return False
+
+        # If this batch can not hold the message in question, return False.
+        if self.size + message.ByteSize() > self.settings.max_bytes:
+            return False
+
+        # Okay, everything is good.
+        return True
+
+    @abc.abstractmethod
+    def publish(self, message):
+        """Publish a single message.
+
+        Add the given message to this object; this will cause it to be
+        published once the batch either has enough messages or a sufficient
+        period of time has elapsed.
+
+        This method is called by :meth:`~.PublisherClient.publish`.
+
+        Args:
+            message (~.pubsub_v1.types.PubsubMessage): The Pub/Sub message.
+
+        Returns:
+            ~.pubsub_v1.publisher.batch.mp.Future: An object conforming to the
+                :class:`concurrent.futures.Future` interface.
+        """
+        raise NotImplementedError
+
+    class Status(object):
+        """An enum class representing valid statuses for a batch.
+
+        It is acceptable for a class to use a status that is not on this
+        class; this represents the list of statuses where the existing
+        library hooks in functionality.
+        """
+        ACCEPTING_MESSAGES = 'accepting messages'
+        ERROR = 'error'
+        SUCCESS = 'success'
+
+
+class RejectionBatch(object):
```

Comment:
Is there any way you accomplish the goal of this class without having it around? A sentinel object maybe?